cli_tools/common/distro/distro.go

```diff
@@ -29,10 +30,32 @@ const (
 	opensuse = "opensuse"
 	rhel     = "rhel"
 	sles     = "sles"
+	slesSAP  = "sles-sap"
 	ubuntu   = "ubuntu"
 	windows  = "windows"
+
+	archX86 = "x86"
+	archX64 = "x64"
 )
 
+var osFlagExpression = regexp.MustCompile(
+	// Distro is required and is comprised of at least one letter.
+	// There may be two segments, separated by a hyphen.
+	// Examples: ubuntu, sles, sles-sap, windows
+	"^(?P<distro>[a-z]+(?:-[a-z]+)?)" +
+		// Version is required, and is at least one word character.
+		// Examples: 2004, 2008, 2008r2
+		"-(?P<version>\\w+)" +
```

Comment:
\w will match upper and lower case characters, while distro matches only lower case characters. Is this inconsistency needed for any OS version?

Reply:
sgtm. Switched to use `[a-z0-9]`